code/go/0chain.net/smartcontract/minersc/fees.go

```diff
@@ -82,50 +70,50 @@ func (msc *MinerSmartContract) emptyPool(mn *MinerNode,
 	err = msc.deletePoolFromUserNode(pool.DelegateID, mn.ID, pool.ID, balances)
 	return
 }
-
+*/
 // unlock deleted pools
 func (msc *MinerSmartContract) unlockDeleted(mn *MinerNode, round int64,
 	balances cstate.StateContextI) (err error) {
-
-	for id := range mn.Deleting {
-		var pool = mn.Active[id]
-		if _, err = msc.emptyPool(mn, pool, round, balances); err != nil {
-			return common.NewError("pay_fees/unlock_deleted", err.Error())
+	for _, pool := range mn.Pools {
+		if pool.Status == stakepool.Deleting {
+			pool.Status = stakepool.Deleted
+			mn.NumActive--
 		}
-		delete(mn.Active, id)
-		delete(mn.Deleting, id)
 	}
 
 	return
 }
 
 // unlock all delegate pools of offline node
-func (msc *MinerSmartContract) unlockOffline(mn *MinerNode,
-	balances cstate.StateContextI) (err error) {
-
-	mn.Deleting = make(map[string]*sci.DelegatePool) // reset
-
-	// unlock all pending
-	for id, pool := range mn.Pending {
-		if _, err = msc.emptyPool(mn, pool, 0, balances); err != nil {
-			return common.NewError("pay_fees/unlock_offline", err.Error())
+func (msc *MinerSmartContract) unlockOffline(
+	mn *MinerNode,
+	balances cstate.StateContextI,
+) error {
+	for id, pool := range mn.Pools {
+		transfer := state.NewTransfer(ADDRESS, pool.DelegateID, pool.Balance)
+		if err := balances.AddTransfer(transfer); err != nil {
+			return fmt.Errorf("pay_fees/unlock_offline: adding transfer: %v", err)
+		}
+		var err error
+		if mn.NodeType == NodeTypeMiner {
+			err = msc.deletePoolFromUserNode(pool.DelegateID, mn.ID, id, stakepool.Miner, balances)
+		} else {
```

Comment:
Check NodeTypeSharder explicitly to and return error for unknown node type to avoid unexpected behavior.